old/rootfs_provider/docker_rootfs_provider.go

```diff
@@ -55,7 +58,17 @@ func (provider *dockerRootFSProvider) ProvideRootFS(logger lager.Logger, id stri
 
 	imageID, envvars, volumes, err := provider.repoFetcher.Fetch(logger, url, tag)
 	if err != nil {
-		return "", nil, err
+		if url.Host == "" {
+			return "", nil, garden.HttpServerToClientError{
+				fmt.Errorf("could not fetch: %s image from %s registry :%s", url.Path[1:], registry.IndexServerAddress(), err),
+				412, //using http precondition fail for errors to be more user friendly across server and client
+			}
+		} else {
+			return "", nil, garden.HttpServerToClientError{
```

Comment:
It'd be best to avoid tying ourselves to Http in the garden backend interface. It'd be better to have a non-HTTP specific error type defined in garden which we could use (maybe `garden.FailedToFetchError` or similar for example). If garden wants to use a 412 code to represent that error type over REST that's up to garden, but garden-linux shouldn't worry about http error codes (or http in general -- that's garden's job).